File: Dev/unitTest_Engine_cs/IO/StaticFile_Package.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace unitTest_Engine_cs.IO
{
    class StaticFile_Package : EngineTest
    {

        public StaticFile_Package()
            : base(3)
        {
        }

        protected override void OnStart()
        {
            //普通に読み込んだバイナリ
            List<byte> loadData;

            using (FileStream fileStream = new FileStream("Data/Texture/Surface/Tile_Spec.png", FileMode.Open, FileAccess.Read))
            {

                byte[] buf = new byte[fileStream.Length];
                fileStream.Read(buf, 0, buf.Length);
                loadData = new List<byte>(buf);
            }

            //ファイル機能から読み込んだバイナリ
            ace.Engine.File.AddRootDirectories("Data/Texture.pack");
            var staticFile = ace.Engine.File.CreateStaticFile("Surface/Tile_Spec.png");
            var staticFileData = staticFile.ReadAllBytes();

            for (int i = 0; i < loadData.Count; ++i)
            {
                byte byteFromRaw = loadData[i];

                byte byteFromFile = staticFileData[i];

                Assert.Equals(byteFromRaw, byteFromFile);
            }
        }
    }
}

File: Dev/unitTest_Engine_cpp_gtest/File/StaticFile_Package.cpp
#include <ace.h>
#include <gtest/gtest.h>
#include <memory>
#include "../EngineTest.h"

using namespace std;
using namespace ace;

class File_StaticFile_Package : public EngineTest
{
public:
	File_StaticFile_Package(bool isOpenGLMode)
		: EngineTest(ace::ToAString("StaticFile_Package"), isOpenGLMode, 1)
	{

	}

protected:

	void OnStart()
	{
		//普通に読み込んだバイナリ
		BinaryReader reader;
		auto data = GetBinaryData(ace::ToAString("Data/Texture/Sample1.png"));
		reader.ReadIn(data.begin(), data.end());

		//ファイル機能で読み込んだバイナリ
		ace::Engine::GetFile()->AddRootDirectories(ace::ToAString("Data/Texture.pack").c_str());
		auto staticFile = ace::Engine::GetFile()->CreateStaticFile(ace::ToAString("Sample1.png").c_str());
		auto staticFileData = staticFile->ReadAllBytes();

		int cnt = 0;
		while (!reader.IsEmpty())
		{
			int8_t byteFromRaw = reader.Get<int8_t>();

			int8_t byteFromFile = staticFileData[cnt++];

			ASSERT_EQ(byteFromRaw, byteFromFile);
		}
	}
};

ENGINE_TEST(File, StaticFile_Package)